app/scripts/services/membership/membership.js

```diff
@@ -0,0 +1,172 @@
+				1);
+    };
+
+    // NOTE: there is currently no way to just gaet a list of kinds of roles from the api,
+    // immitating an object as if from API
+    var getSubjectKinds = function() {
+                            return [{
+                                kind: 'User',
+                                sortOrder: 1
+                              }, {
+                                kind: 'Group',
+                                sortOrder: 2
+                              }, {
+                                kind: 'ServiceAccount',
+                                sortOrder: 3,
+                                description: 'Service accounts provide a flexible way to control API access without sharing a regular user’s credentials.',
+                                helpLinkKey: 'service_accounts'
+                              }, {
+                              // see <origin>/pkg/authorization/api/types.go
+                                kind: 'SystemUser',
+                                sortOrder: 4,
+                                description: 'System users are virtual users automatically provisioned by the system.',
+                                helpLinkKey: 'users_and_groups'
+                              }, {
+                                kind: 'SystemGroup',
+                                sortOrder: 5,
+                                description: 'System groups are virtual groups automatically provisioned by the system.',
+                                helpLinkKey: 'users_and_groups'
+                              }
+                            ];
+                        };
+
+    var mapSubjectKinds = function(subjectKinds) {
+      return _.reduce(
+                      subjectKinds,
+                      function(result, subjectKind) {
+                        result[subjectKind.kind] = subjectKind;
+                        result[subjectKind.kind].name = subjectKind.kind;
+                        result[subjectKind.kind].subjects = {};
+                        return result;
+                    }, {});
+    };
+
+    var mapRoles = function(roles) {
+      return _.reduce(
+                      roles,
+                      function(result, next) {
+                        result[next.metadata.name] = {
+                          name: next.metadata.name,
+                          description: annotation(next, 'description')
+                        };
+                        return result;
+                      }, {});
+    };
+
+    var mapRolebindingsForUI =  function(rolebindings, roles) {
+      var roleMap = mapRoles(roles);
+      var subjectKinds = mapSubjectKinds(getSubjectKinds());
+      _.each(rolebindings, function(rolebinding) {
+        // each subject, (user, bob; group: hobbits)
+        _.each(rolebinding.subjects, function(subject) {
+          if(!subjectKinds[subject.kind].subjects[subject.name]) {
+            subjectKinds[subject.kind].subjects[subject.name]  = {
```

Comment:
for SAs you could have two subjects with the same name but in different namespaces, those would conflict here